frontend/src/ui/Input.tsx

```diff
@@ -36,19 +36,33 @@ export type TextAreaProps = React.ComponentPropsWithoutRef<"textarea"> & {
 
 /** A styled multi-line text area */
 export const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
-    ({ error = false, ...rest }, ref) => (
-        <textarea
-            ref={ref}
+    ({ error = false, ...rest }, ref) => {
+        const textAreaRef = useRef<HTMLTextAreaElement>(null);
+        useImperativeHandle(ref, () => textAreaRef.current as HTMLTextAreaElement);
+        const initialHeight = textAreaRef.current?.scrollHeight;
```

Comment:
Is this necessary? Can't you just use `ref`?

```suggestion
        const initialHeight = ref.current?.scrollHeight;
```

Reply:
I though so as well and tried this of course, but got an error. The ref in this case appears to be a function, so there is no `current` property on it. I found this [stackoverflow](https://stackoverflow.com/questions/62238716/using-ref-current-in-react-forwardref) question about that issue if you're interested (official react docs just say `The ref can be an object or a function.`, unless I missed something).

Reply:
Ah good to know!